rust/toolchain.bzl

```diff
@@ -23,11 +23,15 @@ def build_rustc_command(ctx, toolchain, crate_name, crate_type, src, output_dir,
   if ar_str.find("libtool", 0) != -1:
     ar = "/usr/bin/ar"
 
+  rpaths = _compute_rpaths(ctx.bin_dir, output_dir, depinfo)
+
   # Construct features flags
   features_flags = _get_features_flags(ctx.attr.crate_features)
 
   return " ".join(
       ["set -e;"] +
+      # If TMPDIR is set but not created, rustc will die.
+      ["if [[ -v TMPDIR ]]; then mkdir -p $TMPDIR; fi;"] +
```

Comment:
"-v" is apparently only available since bash 4.2, and "[[ ]]" is apparently not portable.

Try "if [ ! -z "${TMPDIR+x}" ];" ?